config/initializers/sidekiq.rb

```diff
@@ -38,7 +38,7 @@ def not_past_scheduled_time?(current_time)
   sidekiq_url = ApplicationConfig["REDIS_SIDEKIQ_URL"] || ApplicationConfig["REDIS_URL"]
   # On Heroku this configuration is overridden and Sidekiq will point at the redis
   # instance given by the ENV variable REDIS_PROVIDER
-  config.redis = { url: sidekiq_url }
+  config.redis = { url: sidekiq_url, namespace: ENV["REDIS_NAMESPACE"] }
```

Comment:
I am seeing the namespace on most of our Rails traffic but I am not seeing it for some of our Sidekiq keys and in consequence Sidekiq is not working. Below are a couple of examples. 
```
 "dev_local:views/articles/_full_comment_area:de39d75373c05859b3e87229f9867dcf/whole-comment-area-8-2020-09-18 00:09:23 UTC-true",
 "cron_job:send_email_digest:enqueued",
 "cron_job:sitemap_refresh:enqueued",
 "dev_local:stat:failed",
 "queues",
 "cron_job:award_sixteen_week_streak_badge:enqueued",
 "uniquejobs:bfd4f2184500998cceb6328c5dbe617a:GRABBED",
 "cron_jobs",
 "uniquejobs:cdef8e603a2ce66c9c194fefa72687e5:EXISTS",
```

Reply:
This is probably because we'd have to configure both the client and the server connections, here we're only configuring the server, not the client https://dev.to/rbglod/redis-namespace-when-one-sidekiq-process-is-not-enough-39am